api/src/test/java/com/github/bsideup/liiklus/records/LiiklusCloudEventTest.java

```diff
@@ -0,0 +1,101 @@
+package com.github.bsideup.liiklus.records;
+
+import org.assertj.core.data.MapEntry;
+import org.junit.Test;
+import org.junit.experimental.runners.Enclosed;
+import org.junit.runner.RunWith;
+
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.entry;
+
+/**
+ * See <a href="https://github.com/cloudevents/spec/blob/v1.0/kafka-protocol-binding.md">Kafka Protocol Binding</a>
+ */
+@RunWith(Enclosed.class)
+public class LiiklusCloudEventTest {
+
+    public static class Attributes {
+
+        @Test
+        public void serialization() {
+            LiiklusCloudEvent event = newBuilder()
+                    .mediaType("text/plain")
+                    .build();
+
+            assertThat(event.getHeaders()).containsOnly(
+                    entry("ce_specversion", "1.0"),
+                    entry("ce_id", event.getId()),
+                    entry("ce_type", event.getType()),
+                    entry("ce_source", event.getRawSource()),
+                    entry("ce_datacontenttype", "text/plain")
+            );
+        }
+
+        @Test
+        public void deserialization() {
+            String id = UUID.randomUUID().toString();
+            Map<String, String> headers = Stream.of(
```

Comment:
why not just Map.of?

Reply:
`api` is Java 8 for better compatibility, didn't bother to make tests run 11, just in case